### DISCUSSION TITLE
Submodules

### DISCUSSION THREAD

**Discussion by markdav-is** [markdav-is/Hexstor#1] [General]
This repo relies on two submodules: One from inkle/ink and the other from oqtane/oqtane.framework.  In the gitmodules file you can see we are referencing forks maintained in markdav-is.  These forks allow us to pin to a particual release branch and keep us all on the same page.  Whatever git tools you use, be sure and initialize the submodules and verify that you have cloned the proper branch.  submodules can be tricky at times.  

```
[submodule "oqtane.framework"]
	path = oqtane.framework
	url = https://github.com/markdav-is/oqtane.framework.git
	branch = release/v6.0.1
	ignore = all

[submodule "ink"]
	path = ink
	url = https://github.com/markdav-is/ink.git
	branch = release/v1.2.0
	ignore = all
```

**Comment by DrCord** [comment]
in a normal git workflow you can either clone with the `--recurse-submodules` flag or if you've already cloned the repo you can use `git submomdule init && git submodule update`
